lib/ReactViews/Tools/DiffTool/DiffTool.tsx

```diff
@@ -645,6 +810,87 @@ const DatePanel = styled(Box).attrs({
   }
 `;
 
+const AreaFilterSelection = (props: {
+  location?: LatLonHeight;
+  isPickingNewLocation: boolean;
+}) => {
+  const { location, isPickingNewLocation } = props;
+  let locationText = "-";
+  if (location) {
+    const { longitude, latitude } = prettifyCoordinates(
+      location.longitude,
+      location.latitude,
+      {
+        digits: 2
+      }
+    );
+    locationText = `${longitude} ${latitude}`;
+  }
+  const progressComponent = (
+    <Text textLight extraExtraLarge bold>
+      <Loader message={`Querying ${location ? "new" : ""} position...`} />
+    </Text>
+  );
+
+  return (
+    <Box column centered>
+      <Text textLight semiBold>
+        Area filter selection
+      </Text>
+      {isPickingNewLocation && progressComponent}
+      {!isPickingNewLocation && (
+        <Text
+          textLight
+          extraExtraLarge
+          bold
+          css={"min-height: 40px; line-height: 40px;"}
+        >
+          {locationText}
+        </Text>
+      )}
+    </Box>
+  );
+};
+
+const LocationAndDatesDisplayBox = styled(Box).attrs({
+  column: true,
+  charcoalGreyBg: true
+})`
+  color: ${p => p.theme.textLight};
+  padding: 15px;
+  > ${Box}:first-child {
+    margin-bottom: 13px;
+  }
+  > div > div:first-child {
+    /* The labels */
+    margin-right: 10px;
+    min-width: 57px;
+  }
+`;
+
+const LegendImage = function(props: any) {
+  return (
+    <img
+      {...props}
+      // Show the legend only if it loads successfully, so we start out hidden
+      style={{ display: "none", marginTop: "4px" }}
+      // @ts-ignore
+      onLoad={e => (e.target.style.display = "block")}
+      // @ts-ignore
+      onError={e => (e.target.style.display = "none")}
+    />
+  );
+};
+
+const LinkButton = styled(Button)`
```

Comment:
Just a FYI `RawButton` does exactly what you've done here (no need to change, I've swapped it out in my ui branch)